synapse/util/async_helpers.py

```diff
@@ -461,30 +449,35 @@ def cb(_r: None) -> "defer.Deferred[None]":
             # In order to break the cycle, we add a cheeky sleep(0) here to
             # ensure that we fall back to the reactor between each iteration.
             #
-            # (This needs to happen while we hold the lock, and the context manager's exit
-            # code must be synchronous, so this is the only sensible place.)
-            return self._clock.sleep(0)
+            # (This needs to happen while we hold the lock, and the context manager's
+            # exit code must be synchronous, so this is the only sensible place.)
```

Comment:
it's worth noting this is no longer strictly true - the context manager's `__aexit__` doesnt have to be synchronous, so the sleep could happen there instead.

Not sure it's worth changing, just noting that the comment is incorrect.

Reply:
Good spot! I'll update the comment.